src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -710,6 +710,9 @@ trait Namers extends MethodSynthesis {
       if (isScala) {
         if (nme.isSetterName(tree.name)) ValOrVarWithSetterSuffixError(tree)
         if (tree.mods.isPrivateLocal && tree.mods.isCaseAccessor) PrivateThisCaseClassParameterError(tree)
+        if (tree.mods.isImplicit && !owner.isTerm && settings.warnImplicitNoAnnotation && tree.tpt.isEmpty) {
+          reporter.warning(tree.pos, s"implicit value ${tree.name} has no type annotation")
```

Comment:
(I guess this is to avoid overloading the term "annotation" with `@annotations`.)